xray/commands/audit/npm.go

```diff
@@ -0,0 +1,185 @@
+	auditCmd.typeRestriction = typeRestriction
+	return auditCmd
+}
+
+func (auditCmd *AuditNpmCommand) SetServerDetails(server *config.ServerDetails) *AuditNpmCommand {
+	auditCmd.serverDetails = server
+	return auditCmd
+}
+
+func (auditCmd *AuditNpmCommand) ServerDetails() (*config.ServerDetails, error) {
+	return auditCmd.serverDetails, nil
+}
+
+func (auditCmd *AuditNpmCommand) SetWatches(watches []string) *AuditNpmCommand {
+	auditCmd.watches = watches
+	return auditCmd
+}
+
+func (auditCmd *AuditNpmCommand) SetProject(project string) *AuditNpmCommand {
+	auditCmd.projectKey = project
+	return auditCmd
+}
+
+func (auditCmd *AuditNpmCommand) SetTargetRepoPath(repoPath string) *AuditNpmCommand {
+	auditCmd.projectKey = repoPath
+	return auditCmd
+}
+
+func (auditCmd *AuditNpmCommand) SetIncludeVulnerabilities(include bool) *AuditNpmCommand {
+	auditCmd.includeVulnerabilities = include
+	return auditCmd
+}
+
+func (auditCmd *AuditNpmCommand) SetIncludeLincenses(include bool) *AuditNpmCommand {
+	auditCmd.includeLincenses = include
+	return auditCmd
+}
+
+func NewAuditNpmCommand() *AuditNpmCommand {
+	return &AuditNpmCommand{}
+}
+
+func (auditCmd *AuditNpmCommand) Run() (err error) {
+	typeRestriction := auditCmd.typeRestriction
+
+	currentDir, err := coreutils.GetWorkingDirectory()
+	if err != nil {
+		return err
+	}
+	if auditCmd.workingDirectory == "" {
+		auditCmd.workingDirectory = currentDir
+	} else {
+		err = os.Chdir(auditCmd.workingDirectory)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			err = os.Chdir(currentDir)
+		}()
+	}
+	log.Debug("Working directory set to:", auditCmd.workingDirectory)
+
+	packageInfo, err := coreutils.ReadPackageInfoFromPackageJson(auditCmd.workingDirectory)
+	if err != nil {
+		return err
+	}
+	npmExecutablePath, err := npmutils.FindNpmExecutable()
+	if err != nil {
+		return err
+	}
+	// Calculate npm dependencies
+	dependenciesList, err := npmutils.CalculateDependenciesList(typeRestriction, []string{}, npmExecutablePath, packageInfo.BuildInfoModuleId())
+	if err != nil {
+		return err
+	}
+	// Parse the dependencies into an Xray dependency tree format
+	npmGraph := parseNpmDependenciesList(dependenciesList, packageInfo)
+	xrayManager, err := commands.CreateXrayServiceManager(auditCmd.serverDetails)
+	if err != nil {
+		return err
+	}
+	params := services.NewXrayGraphScanParams()
+	params.Graph = npmGraph
+	params.RepoPath = auditCmd.targetRepoPath
+	params.Watches = auditCmd.watches
+	params.ProjectKey = auditCmd.projectKey
+
+	scanId, err := xrayManager.ScanGraph(params)
+	if err != nil {
+		return err
+	}
+	scanResults, err := xrayManager.GetScanGraphResults(scanId, auditCmd.includeVulnerabilities, auditCmd.includeLincenses)
+	if err != nil {
+		return err
+	}
+	tempDirPath, err := fileutils.CreateTempDir()
+	if err != nil {
+		return err
+	}
+	log.Info("The full scan results are available here: " + tempDirPath)
```

Comment:
This shouldn't be a log message, but an ```fmt.Println``` line.
Let's also move the printing of this line to be after the below ```xrutils.WriteJsonResults```, so that it is not printed in case of errors.
The above also goes for the audit commands.